gpflow/base.py

```diff
@@ -10,35 +10,61 @@
 
 from .config import default_float
 
+if TYPE_CHECKING:
+    from IPython.lib import pretty
+
 DType = Union[np.dtype, tf.DType]
-VariableData = Union[List, Tuple, np.ndarray, int, float]
+VariableData = Union[List, Tuple, np.ndarray, int, float]  # deprecated
 Transform = Union[tfp.bijectors.Bijector]
 Prior = Union[tfp.distributions.Distribution]
 
 
-def _IS_PARAMETER(o):
+TensorType = Union[tf.Tensor, tf.Variable, np.ndarray, "Parameter"]
```

Comment:
```suggestion
TensorType = Union[tf.Tensor, tf.Variable, "Parameter"]
```
`np.ndarray` may work in practice but there are occasional differences (see e.g. #1439)

Reply:
was the suggestion to remove `ndarray` entirely from these types?